components/config-mgmt-service/backend/elastic/runs.go

```diff
@@ -69,6 +71,71 @@ func (es Backend) GetRun(runID string, endTime time.Time) (backend.Run, error) {
 	return run, nil
 }
 
+func (es Backend) GetCheckinCountsTimeSeries(startTime, endTime time.Time,
+	filters map[string][]string) ([]backend.CheckInPeroid, error) {
+	var (
+		dateHistoTag = "dateHisto"
+		mainQuery    = newBoolQueryFromFilters(filters)
+		nodeID       = "node_id"
+	)
+
+	rangeQuery, _ := newRangeQuery(startTime.Format(time.RFC3339),
+		endTime.Format(time.RFC3339), backend.RunEndTime)
+
+	mainQuery = mainQuery.Must(rangeQuery)
+
+	bucketHist := elastic.NewDateHistogramAggregation().Field(backend.RunEndTime).
+		Interval("24h").
+		MinDocCount(0). // needed to return empty buckets
+		ExtendedBounds(
+			startTime.Format(time.RFC3339), endTime.Format(time.RFC3339)). // needed to return empty buckets
+		Format("yyyy-MM-dd'T'HH:mm:ssZ").
+		TimeZone(getTimezoneSoStartAtMidnight(startTime)). // needed start the buckets at the beginning of the day.
+		SubAggregation(nodeID,
+			elastic.NewCardinalityAggregation().Field(backend.Id))
+
+	searchResult, err := es.client.Search().
+		Index(IndexConvergeHistory).
+		Query(mainQuery).
+		Aggregation(dateHistoTag, bucketHist).
+		Do(context.Background())
+	if err != nil {
+		return []backend.CheckInPeroid{}, err
+	}
+
+	numberOfHours := getNumberOfHoursBetween(startTime, endTime)
+	numberOfDays := int(math.Ceil(float64(numberOfHours) / 24.0))
+	checkInPeroids := make([]backend.CheckInPeroid, numberOfDays)
+	dateHistoRes, found := searchResult.Aggregations.DateHistogram(dateHistoTag)
+	if !found {
```

Comment:
If there are no runs for the entire range requested `found` will be false. We do not want to return an error if this happens so we create a collection of empty buckets to return.